packages/terra-folder-tree/src/FolderTree.jsx

```diff
@@ -13,26 +17,66 @@ const propTypes = {
    * List of FolderTree.Items to be displayed as content within the FolderTree.
    */
   children: PropTypes.node,
+  /**
+   * The heading level for the title of the folder tree.
+   */
+  headerLevel: PropTypes.number,
   /**
    * The title of the folder tree.
    */
   title: PropTypes.string.isRequired,
   /**
-   * The heading level for the title of the folder tree.
+   * Callback function for expand all event.
    */
-  headerLevel: PropTypes.number,
+  onExpandAll: PropTypes.func,
+  /**
+   * Callback function for collapse all event.
+   */
+  onCollapseAll: PropTypes.func,
+  /**
+   * @private
+   * intl object programmatically imported through injectIntl from react-intl.
+   * */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
 };
 
 const defaultProps = {
   headerLevel: 3,
 };
 
-const FolderTree = ({ children, title, headerLevel }) => (
+const FolderTree = ({
+  children,
+  headerLevel,
+  title,
+  onExpandAll,
+  onCollapseAll,
+  intl,
+}) => (
   <>
     <ActionHeader
       text={title}
       level={headerLevel}
     />
+    <Toolbar
+      align="end"
+      ariaControls={title}
+      ariaLabel={title}
+    >
+      <Button
+        title={intl.formatMessage({ id: 'Terra.folder-tree.button.expand-all' })}
+        text={intl.formatMessage({ id: 'Terra.folder-tree.button.expand-all-instructions' })}
+        variant="utility"
+        icon={<IconExpandRow />}
+        onClick={onExpandAll}
+      />
+      <Button
+        title={intl.formatMessage({ id: 'Terra.folder-tree.button.collapse-all' })}
+        text={intl.formatMessage({ id: 'Terra.folder-tree.button.collapse-all-instructions' })}
+        variant="utility"
+        icon={<IconCollapseRow />}
+        onClick={onCollapseAll}
+      />
```

Comment:
These buttons should have `aria-controls` attributes with the id of the element with `role="tree"`

Reply:
Updated [7ccf080](https://github.com/cerner/terra-framework/pull/1943/commits/7ccf08099fadf52ee1b4d2123aedf9f4d0906305)

This doesn't seem to change the current screenreader behavior in Voiceover though, is that expected?

Reply:
There isn't widespread support for it yet, just future-proofing.